### PR TITLE
Add gallery of examples using sphinx-gallery

### PR DESCRIPTION
**Pull request by kanderso-nrel**
- [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

Prompted by the [Google Groups discussion](https://groups.google.com/d/msg/pvlib-python/VmKGv6uVxsE/_eHy0O65BAAJ), here's an example sunpath diagram.  Result here: https://external-builds.readthedocs.io/html/pvlib-python/846/solarposition.html

I wasn't sure where to put it -- it's currently a new top-level RST file, but maybe it should go in docs/tutorials or elsewhere?

### CODE REVIEW

**Comment by mikofski** [comment]
Is this relevant: [Sphinx gallery](https://sphinx-gallery.github.io/stable/index.html)?

**Comment by kanderso-nrel** [comment]
> Is this relevant: Sphinx gallery?

Very cool!  Would we use it for a "pvlib cookbook" of example snippets?

**Comment by wholmgren** [comment]
I'm in favor of adding an examples section based on sphinx-gallery. A new top-level page seems like a bigger task and would require a lot more thought.

#329 is relevant, but a bigger topic.

**Comment by kanderso-nrel** [comment]
Here's a gallery with a couple examples -- thoughts? 

https://external-builds.readthedocs.io/html/pvlib-python/846/auto_examples/index.html

**Comment by mikofski** [comment]
What do y'all think about moving the example gallery up near the top of the TOC? My thinking is that it might be a hook to entice people to stay on the site longer, and engage more with pvlib, if they find what they're looking for faster? But it might also be a distraction. Looking at matplotlib and a few other sites, related to #329, I've noticed that documentation and examples are separate sections accessed from a common landing page, whereas our documentation is our landing page, so

**Comment by mikofski** [comment]
@kanderso-nrel these example pages are so exciting! Thank you! I was wondering what you think about adding something to the example gallery related to your [secret shores app](../pull/823#issuecomment-570948657)?

**Comment by wholmgren** [comment]
> What do y'all think about moving the example gallery up near the top of the TOC?

I'm fine with this. It also makes me realize that it's going to be confusing to have `Intro Examples` and `Example Gallery`. I'm not sure what to recommend. 

> I was wondering what you think about adding something to the example gallery related to your secret shores app?

Future work. We need to get a bug fix out sooner than later for #844.

**Comment by kanderso-nrel** [comment]
> I was wondering what you think about adding something to the example gallery related to your secret shores app?
> Future work.

Yes I'd like to add something related to backtracking on slopes, but I'm fine with leaving it until later.  For future reference, Bokeh has developed some custom sphinx extensions for embedding interactive bokeh elements in sphinx docs.  It would be a little tricky because the plots are JS-only (i.e. no realtime python backend), so all needed data would have to be generated at compile time, not run time.  Also not 100% sure how it would work with sphinx-gallery.  Relevant links:
- https://docs.bokeh.org/en/latest/docs/reference/sphinxext.html#module-bokeh.sphinxext.bokeh_plot
- https://docs.bokeh.org/en/latest/docs/user_guide/interaction/callbacks.html#javascript-callbacks
- https://github.com/bokeh/docs.bokeh.org/blob/master/en/1.3.4/_sources/docs/user_guide/interaction/callbacks.rst.txt

>  It also makes me realize that it's going to be confusing to have Intro Examples and Example Gallery

Referring back to #329, maybe Intro Examples should aim to fill the `Tutorial` niche?  Right now the Example Gallery entries are sort of between `How-to guides` and `Explanation`, though I'm open to rewriting them (not necessarily in this PR) to align more with one than the other.  

> It seems that most of the flake8 warnings are worth addressing. We can overlook E402 in the examples.

Yep I didn't clean up after myself.  I'll make stickler happy later today.  

I'll also move `Example Gallery` to below the `Intro Examples`, and unless anyone objects, I'll rename `Intro Examples` to `Intro Tutorial`?

**Comment by wholmgren** [comment]
> I'll also move Example Gallery to below the Intro Examples, and unless anyone objects, I'll rename Intro Examples to Intro Tutorial?

That sounds good for this PR scope.

**Comment by kanderso-nrel** [comment]
Besides E402, I think the example scripts should also be given a pass for E265 since sphinx-gallery wants section breaks marked with `#%%`.

File: docs/sphinx/source/introtutorial.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
-.. _introexamples:
+.. _introtutorial:
```
Comment (wholmgren) [inline docs/sphinx/source/introtutorial.rst]: update link in `package_overview.rst` line 15

File: docs/sphinx/source/conf.py [hunk 2]
```diff
@@ -324,3 +325,17 @@ def setup(app):
...
+# settings for sphinx-gallery
+sphinx_gallery_conf = {
+    'examples_dirs': ['../../examples'],  # location of gallery scripts
+    'gallery_dirs': ['auto_examples'],  # location of generated output
+    # sphinx-gallery only shows plots from plot_*.py files by default:
+    # 'filename_pattern': '*.py',
+}
+# supress warnings in gallery output
+# https://sphinx-gallery.github.io/stable/configuration.html
+import warnings
```
Comment (wholmgren) [inline docs/sphinx/source/conf.py]: Not a big deal, but I'd put this import up at the top of the file since you're already making another change.

**Pull request merged**